Bio/GenBank/__init__.py

```diff
@@ -1155,12 +1164,14 @@ def location(self, content):
                 else:
                     part_strand = strand
                 try:
-                    loc = _loc(part, self._expected_size, part_strand)
+                    loc = _loc(part, self._expected_size, part_strand,
+                               seq_type=self._seq_type.lower()).parts
+
                 except ValueError as err:
                     print(location_line)
                     print(part)
                     raise err
-                locs.append(loc)
```

Comment:
Why append vs extend? The ``_loc(...)`` function might return either a ``FeatureLocation`` or a ``CompoundLocation``,  yet we want ``locs`` to be a list of ``FeatureLocation`` objects to wrap up as a ``CompoundLocation``? I may be missing something as I've not run the new code yet,

Reply:
There was a small issue with the way this worked. When the improperly wrapped origin was split into two FeatureLocations, it returned a CompoundLocation containing two FeatureLocations. This could result in passing a CompoundLocation to a CompoundLocation object, causing a problem. Essentially it was creating this:

`CompoundLocation(FeatureLocation(x,y,z), CompoundLocation([FeatureLocation(x,y,z), FeatureLocation(x,y,z)]))`

I've used `extend()` for the location strings that were processed in a loop, and collected the FeatureLocation objects in a list that would ultimately go to a new `CompoundLocation` object. You'll note that I've grabbed a list of the FeatureLocation objects returned from `_loc()` using `.parts`. `extend()` is used to iterate over the list of two `FeatureLocation` objects returned when the origin wrapped string is split in two. I'm happy to commend the code to make this more clear.

I realize my explanation isn't very straight forward. Let me know if this doesn't make sense!

Reply:
It was the ``.parts`` that I missed, the updated comments do help I think.